### ISSUE TITLE
Advice needed. Safest way to upgrade router to newer version?

### ISSUE THREAD

**Issue by peterxxl** [Ysurac/openmptcprouter#3634]
## Expected Behavior
Update router safely to latest version. What's the best way?

## Current Behavior
I have been running v0.60beta1-6.1 r0+23789-ce6ad123e7 on the router for 1 year and half without major issue
VPS has been running OpenMPTCProuter Server 0.1028

I see there have been many new releases so I guess it's time to update the software. 
I have already upgraded the VPS to 0.1031 via the command and it seems to be working fine. 

Now how do I update remotely the router?
I don't want to break anything as the router is not easily accesible and would rather not have to physically visit.

It's x86_64 machine but I no longer remember if the original image was ext4 or squashfs and EFI or not? Where could this information be found? 

Is sysugrade menu the best way to do this? The proposed version doesn't match the latest version in downloads section.
<img width="597" alt="image" src="https://github.com/user-attachments/assets/9810a6ed-ad13-41d2-9b37-52f525881b6b">

or rather this one? as mentioned, for this one I don't remember which image I have used originally
<img width="521" alt="image" src="https://github.com/user-attachments/assets/ac0e8d35-a0f7-4e5b-ae56-437402d65168">

## Specifications

  - OpenMPTCProuter version: v0.60beta1-6.1 r0+23789-ce6ad123e7
  - OpenMPTCProuter VPS version: 0.1031
  - OpenMPTCProuter VPS provider: Grupo ZGH
  - OpenMPTCProuter platform: x86_64

**Comment by Ysurac** [comment]
It seems I forgot to update the version for sysupgrade menu as we have v0.61 now. I will purge CDN cache.
You can update via flash image menu, it's using same program to update in fact.
To check filesystem, via SSH on the router, ```mount | awk 'NR==1{print $5}}'``` and for efi ```ls /sys/firmware/efi``` return some data if it's EFI.

**Comment by peterxxl** [comment]
Hi and thanks for reply... based on your commands, I have determined it's ext4 with EFI.

Unfortunately first & second try didn't work. 
When using `Flash new firmware image` and selecting `openmptcprouter-v0.61-6.6-r0+27346-c7ba5574f5-x86-64-generic-ext4-combined-efi.img.gz` it got stuck on  30.51%
<img width="661" alt="Screenshot 2024-11-03 at 10 55 07 PM" src="https://github.com/user-attachments/assets/7f781afc-69a9-4786-9c7f-0d2c995b40a3">

on second try I used `Sysupgrade` option and got this
<img width="1247" alt="image" src="https://github.com/user-attachments/assets/820afe91-133a-4ea3-96bd-37becedd356c">

**Comment by peterxxl** [comment]
3rd time got stuck on 41.38%
<img width="617" alt="image" src="https://github.com/user-attachments/assets/af41079d-a7b3-44fc-81fe-1c34a24260d6">

**Comment by peterxxl** [comment]
one more question... the image should be uploaded as downloaded `img.gz` file or decompressed as `img` only?
the reason for asking is that the upload is painfully slow, especially when uploading decompressed (around 500MB)
I am doing this from machine on the same LAN via remote desktop, so I guess it should be fast

![image](https://github.com/user-attachments/assets/a95333ce-48b9-4896-b39b-a6a9664d1eba)

**Comment by Ysurac** [comment]
The image should be uploaded to the router as .img.gz. No need to decompress it.
If upload is slow then you have a LAN network issue I think. it' a VM ?

**Comment by peterxxl** [comment]
nope, physical machine bought from Amazon...and  it's connected to LAN via 2.5GbE, so it's really weird why it would take so long... will investigate further.. thanks

**Comment by peterxxl** [comment]
is there a command to preform the same action `Flash new firmware image` via SSH if I download the `.img.gz` file directly to router via `wget`?

**Comment by peterxxl** [comment]
so I did `sysupgrade -v openmptcprouter-v0.61-6.6-r0+27346-c7ba5574f5-x86-64-generic-ext4-combined-efi.img` and seems to have worked... the router booted up after a minute or so with same configuration 

now, I will check if everything works.... nice new UI btw. awesome. thanks

**Comment by jihadismail8** [comment]
<img width="1445" height="463" alt="Image" src="https://github.com/user-attachments/assets/b447b00e-9eb8-44db-b8e0-6cc047d419ce" />
this does not make sense !! , the version i have is higher then the one available , its an upgrade or downgrade ?

**Comment by Ysurac** [comment]
It was a CDN problem, solved with a purge of the URL.
